bot/common/threads/points.py

```diff
@@ -57,42 +100,124 @@ async def send(self, message, user_id):
+
+    name = StepKeys.POINTS_CSV_PROMPT.value
+
+    def __init__(self, channel=None):
+        self.channel = channel
+
+    async def send(self, message, user_id):
+        channel = self.channel
+        if message:
+            channel = message.channel
+
+        sent_message = await channel.send(
+            "Would you like a .csv file of your contributions?"
+        )
+        await sent_message.add_reaction(YES_EMOJI)
+        await sent_message.add_reaction(NO_EMOJI)
+
+        return sent_message, None
+
+
+class GetContributionsCsvPropmtEmoji(BaseStep):
+    """Accepts user emoji reaction to if they want a contributions csv"""
+
+    name = StepKeys.POINTS_CSV_PROMPT_EMOJI.value
+    emoji = True
+
+    def __init__(self, user_id, cache):
+        self.user_id = user_id
+        self.cache = cache
+
+    @property
+    def emojis(self):
+        return [YES_EMOJI, NO_EMOJI]
+
+    async def handle_emoji(self, raw_reaction):
+        if raw_reaction.emoji.name in self.emojis:
+            if raw_reaction.emoji.name == NO_EMOJI:
+                await self.cache.delete(str(self.user_id) + "_contribution_rows")
+                return StepKeys.EMPTY_STEP.value, None
+            return StepKeys.POINTS_CSV_PROMPT_ACCEPT.value, None
+        # Throw here?
+        raise Exception("Reacted with the wrong emoji")
+
+
+class GetContributionsCsvPropmtAccept(BaseStep):
+    """Creates a contributions csv and sends to the user on emoji acceptance"""
+
+    name = StepKeys.POINTS_CSV_PROMPT_ACCEPT.value
+
+    def __init__(self, cache, channel=None):
+        self.cache = cache
+        self.channel = channel
+
+    async def send(self, message, user_id):
+        channel = self.channel
+        if message:
+            channel = message.channel
+
+        contributions = await self.cache.get(str(user_id) + "_contribution_rows")
+        contributions = json.loads(contributions)
+
+        csv = build_csv(contributions[0], contributions[1])
+
+        csvFile = File(
+            csv,
+            str(user_id) + "_points.csv",
+            description="A csv file of your points from contributions",
+            spoiler=False,
+        )
+
+        await self.cache.delete(str(user_id) + "_contribution_rows")
+
+        msg = await channel.send(content="Here's your csv!", file=csvFile)
 
-        return None, {"msg": msg}
+        return msg, None
 
 
 class Points(BaseThread):
     name = ThreadKeys.POINTS.value
 
     async def get_steps(self):
-        return Step(
+        display_points_step = Step(
             current=DisplayPointsStep(
                 guild_id=self.guild_id, cache=self.cache, bot=self.bot
             )
-        ).build()
+        )
+
+        # pass a ref to the DisplayPointsStep so we can retrieve the
+        # contributions when generating the csv without going back to
+        # airtable. the date range from the DisplayPointsStep is also
+        # needed
```

Comment:
This comment is a little confusing. I think my above comment on the metadata should fix this

Reply:
yeah this comment is completely irrelevant now, since I removed the code it was commenting but forgot to remove the comment itself.